PV260_Project/PresentationLayer/UI/ConfigUi.cs

```diff
@@ -0,0 +1,84 @@
+﻿using BusinessLayer.DataLoading;
+using BusinessLayer.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace PresentationLayer.UI
+{
+    internal class ConfigUi : BaseUi, IConfigUi
+    {
+        private readonly IDataDownloader _dataDownloader;
+        private readonly IFileUrlService _fileUrlService;
+
+        public ConfigUi(IDataDownloader dataDownloader, IFileUrlService fileUrlService)
+        {
+            _fileUrlService = fileUrlService;
+            _dataDownloader = dataDownloader;
+        }
+
+        public async Task Run()
+        {
+            await GenerateUi(
+                new List<MenuAction>
+                {
+                    new() { Identifier = "1", Description = "Show current file url used for automatic downloads", Action = ShowCurrentFileUrl },
+                    new() { Identifier = "2", Description = "Set file url for automatic downloads", Action = SetFileUrl },
+                    new() { Identifier = "3", Description = "List all file urls", Action = ListFileUrls },
+                });
+        }
+
+        private async Task ShowCurrentFileUrl()
+        {
+            var fileUrl = await _fileUrlService.GetLatest();
+            Console.WriteLine($"Currently set file url: {fileUrl?.Url ?? "None"}");
+        }
+
+        private async Task SetFileUrl()
+        {
+            Console.WriteLine("Enter the url for automatic file downloads ('b' for back)");
+            var input = Console.ReadLine();
+
+            var urlRegex = new Regex(@"https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)");
+
+            if (input == "b")
+            {
+                return;
+            }
+
+            while (input == null || !urlRegex.IsMatch(input))

```

Comment:
Another syntactic sugar :D nicer way of a null check is the “is” keyword. So “input is null”